src/main/frontend/components/settings.cljs

```diff
@@ -138,12 +140,272 @@
+          :on-blur update-home-page
+          :on-key-press (fn [e]
+                          (when (= "Enter" (util/ekey e))
+                            (update-home-page e)))}]]]])])
+
+(defn encryption-row [t enable-encryption?]
+  (toggle "enable_encryption"
+          (t :settings-page/enable-encryption)
+          enable-encryption?
+          (fn []
+            (let [value (not enable-encryption?)]
+              (config-handler/set-config! :feature/enable-encryption? value)))))
+
+(defn keyboard-shortcuts-row [t]
+  [:div.it.sm:grid.sm:grid-cols-3.sm:gap-4.sm:items-start
+   [:label.block.text-sm.font-medium.leading-5.opacity-70
+    {:for "customize_shortcuts"}
+    (t :settings-page/customize-shortcuts)]
+   [:div.mt-1.sm:mt-0.sm:col-span-2
+    [:div
+     (ui/button
+      (t :settings-page/shortcut-settings)
+      :class "text-sm p-1"
+      :style {:margin-top "0px"}
+      :on-click
+      (fn []
+        (state/close-settings!)
+        (route-handler/redirect! {:to :shortcut})))]]])
+
+(defn auto-push-row [t current-repo enable-git-auto-push?]
+  (when (string/starts-with? current-repo "https://")
+    (toggle "enable_git_auto_push"
+            "Enable Git auto push"
+            enable-git-auto-push?
+            (fn []
+              (let [value (not enable-git-auto-push?)]
+                (config-handler/set-config! :git-auto-push value))))))
+
+(defn usage-diagnostics-row [t instrument-disabled?]
+  (toggle "usage-diagnostics"
+          (t :settings-page/disable-sentry)
+          (not instrument-disabled?)
+          (fn [] (instrument/disable-instrument
+                  (not instrument-disabled?)))
+          [:span.text-sm.opacity-50 "Logseq will never collect your local graph database or sell your data."]))
+
+(defn clear-cache-row [t]
+  [:div.sm:grid.sm:grid-cols-3.sm:gap-4.sm:items-center.sm:pt-5
+   [:label.block.text-sm.font-medium.leading-5.opacity-70
+    {:for "clear_cache"}
+    (t :settings-page/clear-cache)]
+   [:div.mt-1.sm:mt-0.sm:col-span-2
+    [:div.max-w-lg.rounded-md.sm:max-w-xs
+     (ui/button
+      (t :settings-page/clear)
+      :class "text-sm p-1"
+      :on-click handler/clear-cache!)]]])
+
+(defn version-row [t version]
+  [:div.it.app-updater.sm:grid.sm:grid-cols-3.sm:gap-4.sm:items-start
+   [:label.block.text-sm.font-medium.leading-5.opacity-70
+    (t :settings-page/current-version)]
+   [:div.wrap.sm:mt-0.sm:col-span-2
+    (when (util/electron?) (app-updater))
+    [:span.ver version]]])
+
+(defn developer-mode-row [t developer-mode?]
+  (toggle "developer_mode"
+          (t :settings-page/developer-mode)
+          developer-mode?
+          (fn []
+            (let [mode (not developer-mode?)]
+              (state/set-developer-mode! mode)
+              (and mode (util/electron?)
+                   (if (js/confirm (t :developer-mode-alert))
+                     (js/logseq.api.relaunch)))))
+          [:div.text-sm.opacity-50 (t :settings-page/developer-mode-desc)]))
```

Comment:
This big blob of stuff is mostly not new—the key thing it does is go into the settings modal and extract each row as a separate function to make the code a bit easier to read.

The best way to review this would be to scroll down to where all of these functions are called so you can see them in context, then come back up here to view them. Also, I recommend testing out the behavior locally to make sure I didn't inadvertently break anything when I refactored the code. (I already tested it myself, but there's always a chance I missed something!)